src/components/shared/MarkerFiltersContextMenu.js

```diff
@@ -13,23 +13,35 @@ import {
   getMarkersSearchString,
   getSelectedThreadsKey,
 } from 'firefox-profiler/selectors/url-state';
+import { getIsMarkerFiltersMenuVisible } from 'firefox-profiler/selectors/app';
 import { addTransformToStack } from 'firefox-profiler/actions/profile-view';
+import { setMarkerFiltersMenuVisibility } from 'firefox-profiler/actions/app';
 
 import type { ThreadsKey } from 'firefox-profiler/types';
 import type { ConnectedProps } from 'firefox-profiler/utils/connect';
 
 type StateProps = {|
   +searchString: string,
   +threadsKey: ThreadsKey,
+  +isMarkerFiltersMenuVisible: boolean,
 |};
 
 type DispatchProps = {|
   +addTransformToStack: typeof addTransformToStack,
+  +setMarkerFiltersMenuVisibility: typeof setMarkerFiltersMenuVisibility,
 |};
 
 type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
 
 class MarkerFiltersContextMenuImpl extends PureComponent<Props> {
+  _onShow = () => {
+    this.props.setMarkerFiltersMenuVisibility(true);
+  };
+
+  _onHide = () => {
+    this.props.setMarkerFiltersMenuVisibility(false);
+  };
+
```

Comment:
I've been thinking some more during the night (😴), I think it should be possible to just pass it on to the parent (add `onShow` / `onHide` props to `MarkerFiltersContextMenuImpl`, keep a local state in `MarkerSettings`) instead of using the redux state. We don't use this state elsewhere so it's better to keep it local IMO.
 What do you think?

Reply:
It works for me too. I initially wanted to keep that in the redux state thinking that we might need to change it other places too, but this sounds simpler. I changed the code to keep that in local state.